src/main/java/bi/deep/ExactDistinctCountAggregator.java

```diff
@@ -63,8 +63,15 @@ public void aggregate() {
             }
         }
 
-        set.add(getCurrentObjectHashCode(selector));
+        add(set, selectors);
+    }
+
+    public static boolean contains(Set<Integer> set, List<DimensionSelector> selectors) {
+        return selectors.stream().map(ExactDistinctCountAggregator::getCurrentObjectHashCode).allMatch(set::contains);
+    }
 
+    public static void add(Set<Integer> set, List<DimensionSelector> selectors) {
+        selectors.forEach(selector -> set.add(getCurrentObjectHashCode(selector)));
```

Comment:
I have already tried that but it did not work as expected. The result is not matching with previous version. Currently if you apply duplicate values it treats those as single, for example:

if dimension A results into 10
then A | A also returns in 10.

Let me know if this is expected behaviour or not.

Reply:
The unique result in case of multiple dimensions will be the number of unique rows of the table obtained as a result of SELECT dim1, dim2 FROM table. In your example, the result is expected because the same values are added to the HashSet twice, but if they are different, the result of the aggregation may be bigger than the number of rows in the table

Reply:
Yes, for different dimensions it is aggregating the result like 

if dimension A result into 10 and dimension B result into 20 then final result of A and B is 30.